### PR TITLE
[Brave News]: Notify tabs when there are updates to their feed.

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/26631

[Screencast from 2022-11-22 12-45-50.webm](https://user-images.githubusercontent.com/7678024/203180683-11c21dd9-d3d6-49e9-a088-be2916ebd9ae.webm)

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Unfollow a publisher
2. The `Load new content` button should become visible.
3. Refollow the publisher (the button should disappear).
4. Unfollow and click the `Load new content` button. New content should be loaded and the button should disappear.

### CODE REVIEW

File: components/brave_new_tab_ui/api/brave_news/feedListener.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// you can obtain one at https://mozilla.org/MPL/2.0/.
+import getBraveNewsController from '.'
+import {
+  BraveNewsControllerRemote,
+  FeedListenerInterface,
+  FeedListenerReceiver
+} from '../../../../../out/Component/gen/brave/components/brave_today/common/brave_news.mojom.m'
+
+export const addFeedListener = (listener: (feedHash: string) => void) =>
+  new (class implements FeedListenerInterface {
```
Comment (sangwoo108) [inline components/brave_new_tab_ui/api/brave_news/feedListener.ts:13]: Wow, didn't know that there's a class expression like this 👍

File: components/brave_new_tab_ui/api/brave_news/feedListener.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
+import getBraveNewsController from '.'
```
Comment (sangwoo108) [inline components/brave_new_tab_ui/api/brave_news/feedListener.ts:5]: nit: break line?
```suggestion

import getBraveNewsController from '.'
```

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: components/brave_new_tab_ui/api/brave_news/feedListener.ts (hunk 1 above)
Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/api/brave_news/feedListener.ts:14]: Yeah, it's an anonymous class. JavaScript is full of funky features like this.

One that terrifies me is nested destructuring
```js
let { foo: { bar: { hello: result = 7 } } } = { foo: { bar: { hello: 'world' } } }
```

**Pull request merged**